src/view/com/lightbox/ImageViewing/components/ImageItem/ImageItem.android.tsx

```diff
@@ -42,7 +38,112 @@ type Props = {
   doubleTapToZoomEnabled?: boolean
 }
 
-const AnimatedImage = Animated.createAnimatedComponent(Image)
+ type Matrix3 = readonly [
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number,
+  number
+];
+
+function createTransform(): Matrix3 {
+  'worklet'
+  return [1, 0, 0, 0, 1, 0, 0, 0, 1];
+}
+
+function prependTranslate(t: Matrix3, x: number, y: number) {
+  "worklet";
+  t[2] += t[0] * x + t[1] * y;
+  t[5] += t[3] * x + t[4] * y;
+}
+
+function prependScale(t: Matrix3, value: number) {
+  'worklet'
+  t[0] *= value;
+  t[1] *= value;
+  t[3] *= value;
+  t[4] *= value;
+}
+
+function prependTransform(ta: Matrix3, tb: Matrix3) {
```

Comment:
I do not know if this is matrix multiplication. But if it is, I would suggest you should use https://github.com/wcandillon/react-native-redash/blob/master/src/Matrix3.ts#L262. Then you do not need to have this code. Not only this, it has scaleMatrix and translation matrix.

![image](https://github.com/bluesky-social/social-app/assets/5212215/eed3cd2d-ba29-42a4-8287-bb576e212289)

Reply:
I’ve looked at it for inspiration and intentionally chose against it. I don’t want the extra bundle size on the web (and I don’t trust tree shaking to be reliable). I’d also like to avoid any unnecessarily allocations in the gesture handling code. Even if it’s fast, seems unnecessary to add GC churn for things that are enough to mutate in place. The version you linked to always allocates.